### PR TITLE
Prepare for a stable, null-safe v3 release

### PR DESCRIPTION
**Pull request by kevmoo**
- merged prerelease entries in CHANGELOG
- bumped dependencies to stable versions
- LOTS of misc spelling fixes

### CODE REVIEW

**Comment by kevmoo** [comment]
@srawlins @cbracken @jtmcdole ?

File: CHANGELOG.md [hunk 1]
```diff
@@ -41,6 +26,10 @@
   * Deprecate `firstNonNull`. Users of this function should migrate to
     `var v = o1 ?? o2 ?? o3 ?? o4; ArgumentError.checkNotNull(v);`. This will be
     removed in 4.0.0
+  * Eliminate dependency on package:meta.
+  * Fix: `TreeSet.last` was incorrectly returning the minimum node (same
+    value as `TreeSet.first`). It now correctly returns the maximum
+    node.
```
Comment (cbracken) [inline CHANGELOG.md:32]: This last bug never existed in 2.1.5 -- it was introduced in 3.0.0-nullsafety and fixed prior to 3.0.0.

**Review by cbracken — APPROVED** [review]
One nit. Other than that, lgtm. Thanks!

**Comment by cbracken** [comment]
vm_test/dart2js tests are currently failing with this change due to issues with web_socket_channel. Is that expected?

File: CHANGELOG.md (hunk 1 above)
Comment (kevmoo) [reply CHANGELOG.md:32]: ack!

**Comment by kevmoo** [comment]
> vm_test/dart2js tests are currently failing with this change due to issues with web_socket_channel. Is that expected? If so, when do we expect that to be resolved in terms of PR presubmits?

I think I found the issue in the test script. Trying that out!

**Comment by kevmoo** [comment]
@cbracken @srawlins would one of ya'll merge this? looks like I don't have the power

**Comment by kevmoo** [comment]
...or maybe @yjbanov ?

**Comment by cbracken** [comment]
Merged! Safe to publish yet, or should we be holding off until we get word?

**Comment by kevmoo** [comment]
Publish away! You should be good.

You'll get a warning about doing a stable release with a pre-release SDK –
but that's okay this time.

**Pull request merged**